src/js/panels.js

```diff
@@ -198,16 +193,377 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
         }
     });
 
+    /*
+     * language panel
+     */
+    fluid.defaults("gpii.firstDiscovery.panel.lang", {
+        gradeNames: ["fluid.prefs.panel", "{that}.options.prefsEditorConnection", "autoInit"],
+        preferenceMap: {
+            "gpii.firstDiscovery.language": {
+                "model.lang": "default",
+                "controlValues.lang": "enum",
+                "stringArrayIndex.lang": "label",
+                "stringArrayIndex.tooltip": "tooltip",
+                "stringArrayIndex.tooltipAtSelect": "tooltipAtSelect"
+            }
+        },
+        components: {
+            attachTooltipOnLang: {
+                type: "gpii.firstDiscovery.panel.lang.attachTooltipOnLang",
+                container: "{lang}.container",
+                options: {
+                    selectors: "{lang}.options.selectors",
+                    modelRelay: {
+                        source: "{lang}.model.lang",
+                        target: "currentSelectedIndex",
+                        singleTransform: {
+                            type: "fluid.transforms.free",
+                            args: {
+                                "langs": "{lang}.options.controlValues.lang",
+                                "currentLang": "{lang}.model.lang"
+                            },
+                            func: "gpii.firstDiscovery.panel.lang.getCurrentSelectedIndex"
```

Comment:
This function should be turned into a general transform with a generic name, since it will be used very frequently - something like fluid.transforms.indexOf - with the names of the arguments similarly generified - array and element.